### PR TITLE
OAuth: Fix fallback to auto_assign_org_role setting for Azure AD OAuth when no role claims exists

### LINKED ISSUE #30555: Oauth users not being assigned correct role with environment variable GF_USERS_AUTO_ASSIGN_ORG_ROLE
**What happened**:

I'm deploying grafana through Docker 100% as code, no configuration after deploying so the grafana instance and all of its config can be deployed through a pipeline. I'm using environment variables to set all the config options, but one doesn't seem to be working. I'm seeing GF_USERS_AUTO_ASSIGN_ORG_ROLE mentioned in the output logs, but new users who sign in with oauth aren't assigned as "Editor", but "Viewer" instead. I've tried setting this to Admin too, no luck. I've tried with single quotes around 'Editor', but that doesn't work either.

Below is redacted example of my Docker run command:

```
sudo docker run -d \
  --restart=always \
  --user 1000 \
  --net=host \
  --name=grafana \
  -v /data/grafana/var/lib/grafana:/var/lib/grafana \
  -v /data/certs:/data/certs \
  -v /data/grafana/etc/grafana/provisioning:/etc/grafana/provisioning \
  -e "HTTP_PROXY=${MY_PROXY_INFO}" \
  -e "HTTPS_PROXY=${MY_PROXY_INFO}" \
  -e "NO_PROXY=${MY_NOPROXY_INFO}" \
  -e "GF_DASHBOARDS_DEFAULT_HOME_DASHBOARD_PATH=/etc/grafana/provisioning/dashboards/home.json" \
  -e "GF_SERVER_PROTOCOL=https" \
  -e "GF_SERVER_CERT_FILE=/data/certs/cert.crt" \
  -e "GF_SERVER_CERT_KEY=/data/certs/key.key" \
  -e "GF_SERVER_ENABLE_GZIP=true" \
  -e "GF_SERVER_ROOT_URL=https://${MY_GRAFANA_URL}" \
  -e "GF_SERVER_DOMAIN=${MY_GRAFANA_URL}" \
  -e "GF_SECURITY_ADMIN_USER=grafana_admin" \
  -e "GF_SECURITY_ADMIN_PASSWORD=${random_id.grafana_admin_password.hex}" \
  -e "GF_AUTH_AZUREAD_NAME=Azure AD" \
  -e "GF_AUTH_AZUREAD_ENABLED=true" \
  -e "GF_AUTH_AZUREAD_CLIENT_ID=${var.azure_ad_oauth_id}" \
  -e "GF_AUTH_AZUREAD_CLIENT_SECRET=${var.azure_ad_oauth_secret}" \
  -e "GF_AUTH_AZUREAD_SCOPES=openid email profile" \
  -e "GF_AUTH_AZUREAD_AUTH_URL=https://${MY_AZUREAD_URL}/authorize" \
  -e "GF_AUTH_AZUREAD_TOKEN_URL=https://${MY_AZUREAD_URL}/token" \
  -e "GF_USERS_ALLOW_ORG_CREATE=false" \
  -e "GF_USERS_AUTO_ASSIGN_ORG=true" \
  -e "GF_USERS_AUTO_ASSIGN_ORG_ROLE=Editor" \
  -e "GF_USERS_VIEWERS_CAN_EDIT=true" \
  -e "GF_USERS_EDITORS_CAN_ADMIN=true" \
  grafana/grafana:7.3.6
```

I do see these options popping up in the docker logs for this container indicating the options are at least getting passed along:

```
t=2021-01-22T16:02:05+0000 lvl=info msg="Config overridden from Environment variable" logger=settings var="GF_USERS_ALLOW_ORG_CREATE=false"
t=2021-01-22T16:02:05+0000 lvl=info msg="Config overridden from Environment variable" logger=settings var="GF_USERS_AUTO_ASSIGN_ORG=true"
t=2021-01-22T16:02:05+0000 lvl=info msg="Config overridden from Environment variable" logger=settings var="GF_USERS_AUTO_ASSIGN_ORG_ROLE=Editor"
t=2021-01-22T16:02:05+0000 lvl=info msg="Config overridden from Environment variable" logger=settings var="GF_USERS_VIEWERS_CAN_EDIT=true"
t=2021-01-22T16:02:05+0000 lvl=info msg="Config overridden from Environment variable" logger=settings var="GF_USERS_EDITORS_CAN_ADMIN=true"
```

**What you expected to happen**:

Expected with GF_USERS_AUTO_ASSIGN_ORG_ROLE set to Editor and oauth configured and working that new users would be added as Editors, but all new oauth users are added as Viewer instead.

**How to reproduce it (as minimally and precisely as possible)**:

Always tough to debug issues related to oauth... Need a working oauth setup and then set GF_USERS_AUTO_ASSIGN_ORG_ROLE to something other than Viewer.

**Anything else we need to know?**:

**Environment**:
- Grafana version: 7.3.6
- Data source type & version: N/A -- grafana core config
- OS Grafana is installed on: Running in a docker container -- the official docker image unmodified (grafana/grafana:7.3.6)
- User OS & Browser: Mac OS & Chrome
- Grafana plugins: None
- Others: N/A

### PR DESCRIPTION
**Pull request by idafurjes**
**What this PR does / why we need it**:
Rather than fallback to Viewer role when no role claims exists, fixes so it fallback to auto_assign_org_role setting.

**Which issue(s) this PR fixes**:
Fixes #30555 

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/login/social/azuread_oauth.go [hunk 1]
```diff
@@ -97,7 +98,11 @@ func extractEmail(claims azureClaims) string {
 
 func extractRole(claims azureClaims) models.RoleType {
 	if len(claims.Roles) == 0 {
-		return models.ROLE_VIEWER
+		if setting.AutoAssignOrgRole == "" {
```
Comment (marefr) [inline pkg/login/social/azuread_oauth.go:101]: I do believe this is always set and couldn't be empty so no need for this check.

**Review by marefr — COMMENTED** [review]
Looks good, but left a comment.

File: pkg/login/social/azuread_oauth.go (hunk 1 above)
Comment (idafurjes) [reply pkg/login/social/azuread_oauth.go:101]: I see, will fix it

**Review by marefr — APPROVED** [review]
Great. LGTM

**Pull request merged**